parquet/src/test/java/org/apache/iceberg/parquet/TestVariantReaders.java

```diff
@@ -900,6 +1176,31 @@ private static GenericRecord record(GroupType type, Map<String, Object> fields)
     return record;
   }
 
+  private static <T> List<GenericRecord> elements(Type shreddedType, List<T> elements) {
+    GroupType elementType =
+        Types.buildGroup(Type.Repetition.REQUIRED)
+            .addField(
+                Types.primitive(PrimitiveTypeName.BINARY, Type.Repetition.OPTIONAL).named("value"))
+            .addField(shreddedType)
+            .named("element");
+    org.apache.avro.Schema elementSchema = avroSchema(elementType);
+
+    List<GenericRecord> elementRecords = Lists.newArrayList();
+    if (elements != null) {
+      for (T element : elements) {
+        GenericRecord elementRecord = new GenericData.Record(elementSchema);
```

Comment:
Could you use the `record` factory method instead? It would be `record(elementType, Map.of("typed_value", element))`. That way we don't have different ways of constructing these.